Getting `Runtime error: index out of range` on MacOS with VSCode

While running GoDev on MacOS, the following error occasionally occurs:

```
panic: runtime error: index out of range

goroutine 6 [running]:
main.(*WatcherEvent).EventType(...)
        ~/go/src/github.com/zephinzer/godev/watcher.event.go:65
main.(*WatcherEvent).FileType(0xc000350ed8, 0x127678a, 0xc000132070)
        ~/go/src/github.com/zephinzer/godev/watcher.event.go:86 +0x16d
main.(*WatcherEvent).IsAnyOf(0xc000350ed8, 0xc0000984e0, 0x2, 0x2, 0xc000350e01)
        ~/go/src/github.com/zephinzer/godev/watcher.event.go:109 +0x73
main.(*Watcher).watchRoutine(0xc00005a1c0, 0xc000097140, 0xc0002e4f00, 0xc000048bc0, 0xc000048bd0)
        ~/go/src/github.com/zephinzer/godev/watcher.go:91 +0x4a0
created by main.(*Watcher).BeginWatch
        ~/go/src/github.com/zephinzer/godev/watcher.go:64 +0x16b
```

--- a/watcher.event.go
+++ b/watcher.event.go
@@ -30,30 +30,16 @@ const (
 
 var watcherEventType = []string{
 	"",
-	// 1
+	// 00001
 	WatcherEventCreate,
-	// 2
+	// 00010
 	WatcherEventWrite, "",
-	// 4
+	// 00100
 	WatcherEventRemove, "", "", "",
-	// 8
+	// 01000
 	WatcherEventRename, "", "", "", "", "", "", "",
-	//  16
+	// 10000
 	WatcherEventPermission,
-}
-
-var watcherEventTypeString = []string{
-	"",
-	// 1
-	"create",
-	// 2
-	"write", "",
-	// 4
-	"remove", "", "", "",
-	// 8
-	"rename", "", "", "", "", "", "", "",
-	//  16
-	"perms",
 }
 
 // WatcherEvent provides some function candy for working with
@@ -62,12 +48,14 @@ type WatcherEvent fsnotify.Event
 
 // EventType returns a symbol denoting the type of operation recorded
 func (e *WatcherEvent) EventType() string {
-	return watcherEventType[e.Op]
-}
-
-// EventTypeString returns a string denoting the type of operation recorded
-func (e *WatcherEvent) EventTypeString() string {
-	return watcherEventTypeString[e.Op]
+	eventType := ""
+	watcherEvents := []fsnotify.Op{1, 2, 4, 8, 16}
+	for _, event := range watcherEvents {
+		if e.Op|event == event {
+			eventType += watcherEventType[event]
+		}
+	}
+	return eventType
 }
 
 // FilePath returns the absolute path of the file/dir
@@ -83,10 +71,9 @@ func (e *WatcherEvent) FileName() string {
 // FileType returns the extension of the file if its a file,
 // "dir" if its a dir, or "errored" if an error occurred
 func (e *WatcherEvent) FileType() string {
-	switch e.EventType() {
-	case WatcherEventRemove:
+	if e.Op|fsnotify.Remove == fsnotify.Remove {
 		return WatcherFileTypeDeleted
-	default:
+	} else {
 		fileType := path.Ext(e.Name)
 		if len(fileType) == 0 {
 			fileInfo, err := os.Lstat(e.Name)

--- a/watcher.event.go
+++ b/watcher.event.go
@@ -71,23 +71,23 @@ func (e *WatcherEvent) FileName() string {
 // FileType returns the extension of the file if its a file,
 // "dir" if its a dir, or "errored" if an error occurred
 func (e *WatcherEvent) FileType() string {
+	var fileType string
 	if e.Op|fsnotify.Remove == fsnotify.Remove {
-		return WatcherFileTypeDeleted
+		fileType = WatcherFileTypeDeleted
 	} else {
-		fileType := path.Ext(e.Name)
+		fileType = path.Ext(e.Name)
 		if len(fileType) == 0 {
 			fileInfo, err := os.Lstat(e.Name)
 			if err != nil {
-				return WatcherFileTypeErrored
+				fileType = WatcherFileTypeErrored
 			} else if fileInfo.IsDir() {
-				return WatcherFileTypeDir
+				fileType = WatcherFileTypeDir
 			} else {
-				return path.Base(e.Name)
+				fileType = path.Base(e.Name)
 			}
-		} else {
-			return fileType
 		}
 	}
+	return fileType
 }
 
 // IsAnyOf verifies that the file extension matches :theseTypes

--- a/watcher.event_test.go
+++ b/watcher.event_test.go
@@ -61,41 +61,6 @@ func (s *WatcherEventTestSuite) TestEventType_Chmod() {
 	assert.Equal(s.T(), "%", e.EventType())
 }
 
-func (s *WatcherEventTestSuite) TestEventTypeString_Create() {
-	e := WatcherEvent(fsnotify.Event{
-		Op: fsnotify.Create,
-	})
-	assert.Equal(s.T(), "create", e.EventTypeString())
-}
-
-func (s *WatcherEventTestSuite) TestEventTypeString_Write() {
-	e := WatcherEvent(fsnotify.Event{
-		Op: fsnotify.Write,
-	})
-	assert.Equal(s.T(), "write", e.EventTypeString())
-}
-
-func (s *WatcherEventTestSuite) TestEventTypeString_Rename() {
-	e := WatcherEvent(fsnotify.Event{
-		Op: fsnotify.Rename,
-	})
-	assert.Equal(s.T(), "rename", e.EventTypeString())
-}
-
-func (s *WatcherEventTestSuite) TestEventTypeString_Remove() {
-	e := WatcherEvent(fsnotify.Event{
-		Op: fsnotify.Remove,
-	})
-	assert.Equal(s.T(), "remove", e.EventTypeString())
-}
-
-func (s *WatcherEventTestSuite) TestEventTypeString_Chmod() {
-	e := WatcherEvent(fsnotify.Event{
-		Op: fsnotify.Chmod,
-	})
-	assert.Equal(s.T(), "perms", e.EventTypeString())
-}
-
 func (s *WatcherEventTestSuite) TestFilePath() {
 	e := WatcherEvent(fsnotify.Event{
 		Op:   fsnotify.Chmod,

